package.json

```diff
@@ -740,6 +740,20 @@
 							]
 						}
 					}
+				},
+				"haxe.extendedIndentation": {
+					"type": "string",
+					"default": "dynamic",
+					"enum": [
+						"dynamic",
```

Comment:
I don't think we need to have three options here, a simple boolean on-off switch for the `type` command should be enough? VSCode has a built-in setting for disabling the auto-indentation provided by the language configuration via `"editor.autoIndent"`.

Reply:
You mean `indentNextLinePattern` indentation should be here in both cases and can be disabled for Allman style with setting `editor.autoIndent` to `brackets`? Not sure if some features will not be missed in this case (for example `onEnterRules` will be ignored). Also you cannot set `autoIndent` by languages (?), so multilang users will have to change that setting when they switch to another language sometimes.

Reply:
I mean that we never have to change what language configuration we register, only our custom `{` handling needs an off-switch.

There are language specific settings: https://code.visualstudio.com/docs/getstarted/settings#_language-specific-editor-settings

```json
"[haxe]": {
	"editor.autoIndent": "keep"
}
```

Reading through the docs for the `autoIndent` options available, it gives pretty fine-grained control, accounting for both indentation and onEnter rules.

![](https://i.imgur.com/8iGEq0V.png)